src/font.rs

```diff
@@ -107,12 +107,121 @@ impl<'a> Stream<'a> {
+    /// The font variant.
+    pub font_variant: Option<&'a str>,
+    /// The font weight.
+    pub font_weight: Option<&'a str>,
+    /// The font stretch.
+    pub font_stretch: Option<&'a str>,
+    /// The font size.
+    pub font_size: &'a str,
+    /// The font family.
+    pub font_family: &'a str,
+}
+
+/// Split the font shorthand into its respective components.
+pub fn parse_font_shorthand(text: &str) -> Result<FontShorthand, Error> {
+    let mut stream = Stream::from(text);
+    stream.skip_spaces();
+
+    let mut prev_pos = stream.pos();
+
+    let mut font_style = None;
+    let mut font_variant = None;
+    let mut font_weight = None;
+    let mut font_stretch = None;
+
+    for _ in 0..4 {
+        let ident = stream.consume_ascii_ident();
+
+        match ident {
+            // TODO: Reuse actual parsers to prevent duplication.
+            // We ignore normal because it's ambiguous to which it belongs and all
+            // other attributes need to be resetted anyway.
+            "normal" => {}
+            "small-caps" => font_variant = Some(ident),
+            "italic" | "oblique" => font_style = Some(ident),
+            "bold" | "bolder" | "lighter" | "100" | "200" | "300" | "400" | "500" | "600"
+            | "700" | "800" | "900" => font_weight = Some(ident),
+            "ultra-condensed" | "extra-condensed" | "condensed" | "semi-condensed"
+            | "semi-expanded" | "expanded" | "extra-expanded" | "ultra-expanded" => {
+                font_stretch = Some(ident)
+            }
+            _ => {
+                // Not one of the 4 properties, so we backtrack and then start pasing font
+                // size and family.
+                stream = Stream::from(text);
+                stream.advance(prev_pos);
+                break;
+            }
+        }
+
+        stream.skip_spaces();
+        prev_pos = stream.pos();
+    }
+
+    prev_pos = stream.pos();
+    if stream.curr_byte()?.is_digit() {
+        // A font size such as '15pt'
+        let _ = stream.parse_length()?;
+    } else {
+        // A font size like 'xx-large'
+        let size = stream.consume_ascii_ident();
+
+        if !matches!(
+            size,
+            "xx-small"
+                | "x-small"
+                | "small"
+                | "medium"
+                | "large"
+                | "x-large"
+                | "xx-large"
+                | "larger"
+                | "smaller"
+        ) {
+            return Err(Error::UnexpectedData(prev_pos));
+        }
+    }
+
+    let font_size = stream.slice_back(prev_pos);
+    stream.skip_spaces();
+
+    if stream.curr_byte()? == b'/' {
+        // We can ignore line height
+        stream.advance(1);
+        stream.skip_spaces();
+        let _ = stream.parse_length()?;
+        stream.skip_spaces();
+    }
+
+    if stream.at_end() {
+        return Err(Error::UnexpectedEndOfStream);
+    }
+
+    let font_family = stream.slice_tail();
```

Comment:
Shouldn't we parse families here? Or is it planned for later?

Reply:
It will be done on the `resvg` side. We only pass whatever it contains back to `resvg`, which will then "replace" the add the attribute `FontFamily` with the raw string, and once it's processed normally it will get parsed.